### ISSUE TITLE
Not working with api 23. FloatMath is deprecated. use Math instead.

### ISSUE THREAD

**Issue by maxandron** [JakeWharton/ViewPagerIndicator#366]
com.viewpagerindicator.LinePageIndicator: can't find referenced method 'float ceil(float)' in library class android.util.FloatMath

**Comment by beraldofilippo** [comment]
Same here! I hope for a quick fix. In the meantime, is there any workaround?

**Comment by dvdh** [comment]
If you look at the source for FloatMath, it is just doing this:
public static float ceil(float value) {
        return (float) Math.ceil(value);
    }
very straightforward to fix.

**Comment by mturki** [comment]
Same here...

**Comment by HE-LU** [comment]
bump

**Comment by zfoltin** [comment]
+1

**Comment by maxandron** [comment]
For me the current solution is to add the following line to my proguard file:
`-dontwarn com.viewpagerindicator.**`

**Comment by tomafc330** [comment]
If you are not using the Line style indicator then @maxandron's answer works

**Comment by yudhir** [comment]
I am getting the same error for API 23 . All functions were removed!

**Comment by ImranAtBhimsoft** [comment]
Same here.

**Comment by ggrell** [comment]
Fixed in this PR: https://github.com/JakeWharton/ViewPagerIndicator/pull/370 just needs a merge & release

**Comment by ngoohebi** [comment]
+1

**Comment by Shahroz16** [comment]
+1

**Comment by trinadhkoya** [comment]
You  can make use of this (float)Math.sqrt(...)

**Comment by d34n0s** [comment]
+1

**Comment by expilu** [comment]
+1

**Comment by black-snow** [comment]
looking forward to a fix/ merge of the fix + release

**Comment by AntonioCappiello** [comment]
+1

**Comment by trinadhkoya** [comment]
I already mentioned make use of (float)Math.sqrt(...).it will work for you sure.I did it and worked well for me

**Comment by AntonioCappiello** [comment]
@trinadhkoya I actually don't need to use that function, but proguard complains about that when I build the release version and I don't know if the library itself use that function somewhere and may cause some run-time exception

**Comment by trinadhkoya** [comment]
​Sorry  @Antonio

**Comment by calvarez-ov** [comment]
+1

**Comment by lorenzopolidori** [comment]
A more targeted workaround is to add to proguard:
`-dontwarn com.viewpagerindicator.LinePageIndicator`.

**Comment by vincent-paing** [comment]
+1

**Comment by AntonioCappiello** [comment]
Thanks @lorenzopolidori, your suggestion may temporarily hide the problem from the build server log but it doesn't actually solved it as suggested [here](https://github.com/JakeWharton/ViewPagerIndicator/pull/385).

Nobody wants that in "a shiny day close to the release the app breaks because of some deprecated method which disappears from the lib" :-)

**Comment by lorenzopolidori** [comment]
@AntonioCappiello Yes, I agree. The workaround I suggested is just to make a build successful until one of the proposed PR's is accepted and merged into the project (which should happen asap I think).

**Comment by avafab** [comment]
+1

**Comment by jczerski** [comment]
+1

**Comment by adrien-aubel** [comment]
+1

**Comment by ubeyou** [comment]
+1

**Comment by marcinbak** [comment]
+1

**Comment by MostafaSh** [comment]
Hey  guys!
just do what i say to solve the problem. 
replace 
return  (int)floatMath.ceil(result);
with this line:
return (int)((float)Math.ceil(result));

I have tried this solution. It works correctly.

**Comment by AntonioCappiello** [comment]
Hi Mostafa, it is what I did in the PR connected to this issue (https://github.com/JakeWharton/ViewPagerIndicator/pull/385) but it was not merged yet. Probably you haven't read the other comments above. I don't use that function, but the library does. So the warning is not in my code, but in the library. I don't have anything to replace in my code.

**Comment by MostafaSh** [comment]
Hi Antonio.
I know. The error is from the library. I had that error to. You have to modify the library.
I'm using android studio. In android studio you can modify any library simply.
If you are using eclipse, just try this :
Extract the jar file out of eclipse, modify the android.util.FloatMath class as I said, add it to archive again as a jar file and import it to your project.
If you had more questions, you can ask me.

And as a friendly offer, just give up working with eclipse and start working with android studio. you will love it.

**Comment by MostafaSh** [comment]
Ignore my pervious answer.I thought it has a jar file. I will make a video to show you how you can solve the problem and I will  put the video link here .

**Comment by AntonioCappiello** [comment]
ahah did I ever mentioned that I use Eclipse?

**Comment by MostafaSh** [comment]
Watch this video.
https://m.youtube.com/watch?feature=youtu.be&v=I1fGFL2rqaY
If this video wasn't enough, just tell me to make a better one.

**Comment by AntonioCappiello** [comment]
@MostafaSh Why do you include libraries in your project in such "old-fashion" way instead of using a "compile" in gradle pointing to maven/jcenter repo? If the library get updated (bug fixes, improvements, deletion of deprecated methods) then you don't get any of those updates in your way. 
Thanks for making the video anyway. 
I think that the best solution so far is still to merge the PR mentioned before.

**Comment by MostafaSh** [comment]
Actually I'm new in android studio. Thank you for this information about using maven. I don't know anything about that.
But in this situation, for using this library, I succeed with that "old fashion" way!
anyway,

**Comment by MostafaSh** [comment]
I'm looking for more information about using maven.that would be kind of you if you get me more information.

**Comment by jrejaud** [comment]
Hi everyone, I made a fork of ViewPagerIndicator which is compiled via Gradle (thus you can use JitPack) and uses `Math` instead of `FloatMath`, so it works with API 23.

It solved my problem, let me know if it works for you too :)

https://github.com/jrejaud/ViewPagerIndicator2

**Comment by danikula** [comment]
I [published here](https://bintray.com/alexeydanilov/maven/viewpagerindicator) my [solution](https://github.com/danikula/ViewPagerIndicator). It is original Jake's repo with one more additional [commit](https://github.com/danikula/ViewPagerIndicator/commit/3baf609a66b15342014c318c398c1499e4c3d60b) with [this fix](https://github.com/JakeWharton/ViewPagerIndicator/pull/370).

Usage:

```
repositories {
    maven { url 'https://dl.bintray.com/alexeydanilov/maven' }
}
dependencies {
    compile 'com.viewpagerindicator:viewpagerindicator:2.4.3'
}
```

**Comment by luongbeta2** [comment]
+1

**Comment by yakubbaev** [comment]
+1

**Comment by danikula** [comment]
@yakubbaev , @luongbeta2 This repo is not longer supported. There is no chance Jake will fix it and publish. I recommend to use [my solution](https://github.com/JakeWharton/ViewPagerIndicator/issues/366#issuecomment-184176432).

**Comment by luongbeta2** [comment]
thanks

**Comment by dayong38** [comment]
:+1:

**Comment by jrejaud** [comment]
And if you're a `gradle` user instead of `maven`, then I recommend my solution: https://github.com/jrejaud/ViewPagerIndicator2

**Comment by anibalc9** [comment]
+1

**Comment by AbdulRehmanNazar** [comment]
+1
